ui/lib/pki/addon/components/page/pki-configuration-edit.ts

```diff
@@ -43,28 +43,41 @@ export default class PkiConfigurationEditComponent extends Component<Args> {
   @service declare readonly version: VersionService;
 
   @tracked invalidFormAlert = '';
-  @tracked errorBanner = '';
+  @tracked errorBanner: object[] = [];
 
   get isEnterprise() {
     return this.version.isEnterprise;
   }
 
+  async performSave() {
+    for (const modelName of ['cluster', 'acme', 'urls', 'crl']) {
+      const model = this.args[modelName as keyof Args];
+      // only attempt save if user has permission
+      if (!model.canSet) continue;
+      try {
+        await model.save();
+        this.flashMessages.success(`Successfully updated config/${modelName}`);
+      } catch (error) {
+        const errorObject: { modelName: string; message: string } = {
+          modelName,
+          message: errorMessage(error),
+        };
+        this.flashMessages.danger(`Error updating config/${modelName}`, { sticky: true });
+        this.errorBanner.pushObject(errorObject);
```

Comment:
I don't think `pushObject` is necessary here. Did you have issues with change detection when using `push` out of curiosity? Personally I would use the native array methods unless there is a convenience method from Ember that is better suited.

Reply:
Since `this.errorBanner` is a tracked property, `pushObject` was required to get it to update and render the error banner in the DOM

Reply:
Interesting I would have thought it would have detected the change just using `push`.